test/utils/runners.go

```diff
@@ -245,6 +248,12 @@ func (config *DeploymentConfig) create() error {
 		},
 	}
 
+	if len(config.SecretNames) > 0 {
+		volumes, mounts := getVolumesForSecrets(config.SecretNames)
```

Comment:
There are different ways of passing secrets. Volumes are one way. But there are other ways:
- as env vars:
https://github.com/kubernetes/kubernetes/blob/master/pkg/api/types.go#L1095
- image pull secrets:
https://github.com/kubernetes/kubernetes/blob/master/pkg/api/types.go#L1818
We should at least make it possible to use the all. 

So I would suggest changing ```getVolumesForSecrets ``` method to:
```attachSecrets(template *PodTemplateSpec, secretNames []string) ```
or sth like that
[even though in the first try you may want to mount all of them as volumes]